File: patterns/pattern4/pattern4.py
# inverted hollow half pyramid
# * * * * * 
# *     *
# *   *
# * *
# *
n=int(input("enter the value of n : "))
for i in range(n-1,-1,-1):
     for j in range(i+1):
          if(i==n-1 or j==0 or i==0 or j==i):
               print("*",end=" ")
          else:
               print(" ",end=" ")
     print()

File: patterns/pattern4/pattern4.cpp
// inverted hollow half pyramid
// * * * * * 
// *     *
// *   *
// * *
// *
#include<iostream>
using namespace std;
int main(){
     int n;
     cout<<"enter the value of n : ";
     cin>>n;
     for (int i=n;i>0;i--){
          for(int j=0;j<i;j++){
               if(i==n || j==0 || i==j-1 || j==i-1){
                  cout<<"* " ;
               }
               else{
                    cout<<"  ";
               }
          }
          cout<<endl;
     }
     return 0;
}